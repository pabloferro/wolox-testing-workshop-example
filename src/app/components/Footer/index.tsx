import React from 'react';
import i18next from 'i18next';

import styles from './styles.module.scss';

function Footer() {
  return (
    <footer className={`column center middle ${styles.footer}`}>
      <header className="column center m-bottom-8">
        <q cite={`${i18next.t('Home:footerQuoteAuthor')}`} className="base-text fw-bold white-color">
          {i18next.t('Home:footerQuote')}
        </q>
        <span className="base-text fw-bold white-color">{i18next.t('Home:footerQuoteAuthor')}</span>
      </header>
      <small className="small-text white-color fw-thin self-end">
        {i18next.t('Home:madeText')}
        <span className="m-left-1" role="img" aria-label="love">
          ❤️
        </span>
        {i18next.t('Home:madeBy')}
      </small>
    </footer>
  );
}

export default Footer;
