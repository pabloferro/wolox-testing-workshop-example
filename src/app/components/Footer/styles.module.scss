@import 'variables/colors';

.footer {
  background-color: $cod-gray;
  padding: 50px 40px 30px;
}
