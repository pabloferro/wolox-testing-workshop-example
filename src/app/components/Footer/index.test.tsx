import React from 'react';
import { render } from '@testing-library/react';

import Footer from './index';

test('renders correctly', () => {
  const { container } = render(<Footer />);

  expect(container.firstChild).toMatchSnapshot();
});
