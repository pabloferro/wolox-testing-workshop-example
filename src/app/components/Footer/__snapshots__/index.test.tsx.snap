// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  class="column center middle footer"
>
  <header
    class="column center m-bottom-8"
  >
    <q
      cite="Home:footerQuoteAuthor"
      class="base-text fw-bold white-color"
    >
      Home:footerQuote
    </q>
    <span
      class="base-text fw-bold white-color"
    >
      Home:footerQuoteAuthor
    </span>
  </header>
  <small
    class="small-text white-color fw-thin self-end"
  >
    Home:madeText
    <span
      aria-label="love"
      class="m-left-1"
      role="img"
    >
      ❤️
    </span>
    Home:madeBy
  </small>
</footer>
`;
